opensubtitles/opensubtitles.go

```diff
@@ -0,0 +1,286 @@
+	headerAccept      = "Accept"
+	headerUserAgent   = "User-Agent"
+)
+
+// Credentials used to authenticate to make requests to the OpenSubtitles API.
+type Credentials struct {
+	Username string
+	Password string
+}
+
+// A Client manages communication with the GitHub API.
+type Client struct {
+	clientMu sync.Mutex   // clientMu protects the client during calls that modify the CheckRedirect func.
+	client   *http.Client // HTTP client used to communicate with the API.
+
+	// Base URL for API requests. Defaults to the public GitHub API, but can be
+	// set to a domain endpoint to use with GitHub Enterprise. BaseURL should
+	// always be specified with a trailing slash.
+	BaseURL *url.URL
+
+	// Base URL for uploading files.
+	UploadURL *url.URL
+
+	// User agent used when communicating with the GitHub API.
+	UserAgent string
+
+	Credential Credentials
+
+	rateMu sync.Mutex
+
+	Token string
+	//rateLimits [categories]Rate // Rate limits for the client as determined by the most recent API calls.
+
+	common service // Reuse a single struct instead of allocating one for each service on the heap.
+
+	// Services used for talking to different parts of the GitHub API.
+	Authentication *AuthenticationService
+	Discover       *DiscoverService
+	Download       *DownloadService
+	Find           *FindService
+	Info           *InfoService
+	Search         *SearchService
+}
+
+type service struct {
+	client *Client
+}
+
+// addOptions adds the parameters in opts as URL query parameters to s. opts
+// must be a struct whose fields may contain "url" tags.
+func addOptions(s string, opts interface{}) (string, error) {
+	v := reflect.ValueOf(opts)
+	if v.Kind() == reflect.Ptr && v.IsNil() {
+		return s, nil
+	}
+
+	u, err := url.Parse(s)
+	if err != nil {
+		return s, err
+	}
+
+	qs, err := query.Values(opts)
+	if err != nil {
+		return s, err
+	}
+
+	u.RawQuery = qs.Encode()
+	return u.String(), nil
+}
+
+// NewClient returns a new GitHub API client. If a nil httpClient is
+// provided, a new http.Client will be used. To use API methods which require
+// authentication, provide an http.Client that will perform the authentication
+// for you (such as that provided by the golang.org/x/oauth2 library).
+func NewClient(httpClient *http.Client, token string, cred Credentials) (c *Client) {
+	if httpClient == nil {
+		httpClient = &http.Client{}
+	}
+	baseURL, _ := url.Parse(defaultBaseURL)
+	//uploadURL, _ := url.Parse(uploadBaseURL)
+
+	c = &Client{client: httpClient, BaseURL: baseURL, UserAgent: userAgent, Token: token}
+	c.common.client = c
+	c.Authentication = (*AuthenticationService)(&c.common)
+	c.Discover = (*DiscoverService)(&c.common)
+	c.Download = (*DownloadService)(&c.common)
+	c.Find = (*FindService)(&c.common)
+	c.Info = (*InfoService)(&c.common)
+	c.Search = (*SearchService)(&c.common)
+
+	//Check if struct Credential is not empty
+	if (Credentials{}) != cred {
+		c.Credential = cred
+	}
+
+	return
+}
+
+func (c *Client) Connect() (*Client, error) {
```

Comment:
Codacy found an issue: [exported method Client.Connect should have comment or be unexported](https://app.codacy.com/manual/TheForgotten69/go-opensubtitles/pullRequest?prid=6093402)